### PR TITLE
TopNav: New page layouts 

### PR DESCRIPTION
**Pull request by torkelo**
Part of #50791

This explores a way to add new Page layouts behind a feature toggle by replacing the current Page & Page.Contents components. (Two different implementations with same interface that behave differently) 

To make the new nav and old nav work behind the same interface with minimal changes to the actual pages I have to introduce some changes to the Page component 
* Adds navId, this replaces the navModel property (only partially migrated in this PR, I will complete in follow-up PR). This property is used differently in old and new Page components. This navId is used to lookup the nav model from redux navIndex. This replaces the getNavModel found on many pages in mapStateToProps). In the new Page component, it is used for section nav and main page title/header (if no item page nav). 
* Adds pageNav property to Page, this is used for item-level PageHeader in both old and new Page. For the old Page it will override any navId navModel (as the old Page does not use both). Having both navId and pageNav makes it possible to support both navigation modes at the same time with the same Page component interface. 

Note to reviewers
- I can see this going through some big iterations in the next 1-2 month as we deal with more edge cases and plugins and to find room to simplify. It is a bit messy and complex. But the requirements and being backward compatible (and changing as little as possible in each step) make this tricky. But something like this is a first good step unless we want to start a major nav model refactoring. 

Things that need more work in future PRs
* Move all pages that can to use navId or pageNav and remove navModel from Page component 
* Update item level navigation pages (Data sources, users, plugins) to use pageNav instead of navModel   (this should have no change when feature toggle is of but make the page look properly with section nav when feature toggle is on) 

To test. 
* Click through all pages and make sure they work like before (when feature toggle is off). 
* When feature toggle `topnav` is enabled, all root list pages (non item pages) should work ok. The only item page update is this PR is the team page. 
* If you are seeing "Page not found" in breadcrumbs when topnav feature is enabled this expected until all routes have navId added  

TODO 
- [x] Tests for old Page component and behavior 
- [x] Tests for new Page component and behavior

Enterprise PR https://github.com/grafana/grafana-enterprise/pull/3543

Follow up PRs
* https://github.com/grafana/grafana/pull/51664
* https://github.com/grafana/grafana/pull/51666

### CODE REVIEW

**Comment by torkelo** [comment]
Pushed a pretty huge refactor that moves the top navigation and TopNavPage -> renamed to AppChrome to be outside route, this makes it so that moving between pages (routes) never causes an unmount on the app chrome (topnav). 

This refactor also removes the navId from the route and makes updates to the chrome more uniform, navId is now only specified via the Page component (Which bubbles it up the AppChrome via AppChromeService)

File: .betterer.results [hunk 1]
```diff
@@ -3114,6 +3114,12 @@ exports[`better eslint`] = {
       [0, 0, 0, "Do not use any type assertions.", "0"],
       [0, 0, 0, "Do not use any type assertions.", "1"]
     ],
+    "public/app/core/components/PageNew/PageTabs.tsx:5381": [
+      [0, 0, 0, "Do not use any type assertions.", "0"]
+    ],
+    "public/app/core/components/PageNew/SectionNav.tsx:5381": [
+      [0, 0, 0, "Do not use any type assertions.", "0"]
+    ],
```
Comment (ashharrison90) [inline .betterer.results:3122]: this is fine for now 👍 think we'll need to move the `IconName` interface to `@grafana/data` in order to properly fix this? 🤔

File: public/app/routes/routes.tsx [hunk 2]
```diff
@@ -33,7 +33,6 @@ export function getAppRoutes(): RouteDescriptor[] {
     },
     {
       path: '/d/:uid/:slug?',
-      navId: 'dashboards',
```
Comment (ashharrison90) [inline public/app/routes/routes.tsx:36]: so is the idea we don't set the `navId` in the routes and instead set it in the individual components? what are the pros/cons here? 🤔

**Review by ashharrison90 — COMMENTED** [review]
couple of questions as i started looking, will continue reviewing

note for future reviewers: this looks twice as intimidating as it is because ~50 of these changes are updating an import from default -> named

File: public/app/routes/routes.tsx (hunk 2 above)
Comment (torkelo) [reply public/app/routes/routes.tsx:36]: Well it was needed on Page component, so instead of having it twice, it's now only specified there

Comment (torkelo) [reply public/app/routes/routes.tsx:36]: @ashharrison90 

I agree, but passing it down is not that easy without wrapping things in a new React Context or passing router props. And needs to go up and down, as the AppChrome is above, the Page below. So by having it on Page at least for now is the simplest, but we can revisit that as we explore how the Page component will work from plugins and what customization they will need.

File: public/app/core/components/AppChrome/AppChromeService.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+  }
+
+  useState() {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useObservable(this.state, this.state.getValue());
+  }
+}
+
+export const appChromeService = new AppChromeService();
```
Comment (torkelo) [inline public/app/core/components/AppChrome/AppChromeService.tsx:51]: Don't like this global variable here, think we should introduce GrafanaApplicationContext (react context) where we can place services 

But want to keep this PR small and not introduce a big change like that and instead follow the pattern of global service singletons

**Review by joshhunt — APPROVED** [review]

**Pull request merged**